graph/src/data_source/causality_region.rs

```diff
@@ -47,6 +49,15 @@ impl CausalityRegion {
     pub const fn next(self) -> Self {
         CausalityRegion(self.0 + 1)
     }
+
+    pub fn from_entity(entity: &Entity) -> Self {
+        CausalityRegion(
+            entity
+                .get("causality_region")
+                .and_then(|v| v.as_int())
+                .unwrap_or(0),
```

Comment:
```suggestion
                .unwrap_or(CausalityRegion::ONCHAIN),
```

Reply:
covering this with a basic unit test would also be good. If we don't replace the constant it could happen that someone changes the value and then this would be broken.

Reply:
I added this to the `relational_bytes::find` test.